build.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
```

Comment:
My intention with this new script is to aim to retire out `install_analysistools.sh` for two separate scripts:
* `build.sh`
* `install.sh`

The big difference here is that this moves to a paradigm where building is _always_ within the local directory (and so `build/` and `include/` would be in this folder, and that installing just moves the resulting executables to the right place. This has the following benefits:

1. We are more inline with how most other projects function, where it's usually `make` to build and then `make install` to install, where we would have `build.sh` and `install.sh`.
2. This avoids having `build/` and `include/` owned by root on the local machine, while no other folders were, making development annoying
3. We avoid running `make` in tree-sitter as sudo. While we can probably trust the repo to not do anything malicious, I'd rather not give it the chance
4. Environment variables can be acquired from the user scope, and we don't need to export stuff for root to use. This is mostly beneficial on github action, where we can make use of the suggested practices for setting environment variables.